### PR TITLE
Adds function to get state to tracker

### PR DESCRIPTION
**Pull request by skrawcz**
So that we can pull the last state given an ID from the UI.

Updates examples and adds a new one that mirrors the LCEL example with the tool calling. This variant works. Where as the Hamilton DAG also doing tool calling somehow doesn't quite work as well. Still digging in there. Need a run diff tool now... or a way to compare runs...

### CODE REVIEW

File: burr/tracking/client.py [hunk 1]
```diff
@@ -61,12 +62,62 @@ def __init__(
...
+
+    @classmethod
+    def get_state(
+        cls,
+        project: str,
+        app_id: str,
+        sequence_no: int = -1,
+        storage_dir: str = DEFAULT_STORAGE_DIR,
+    ) -> tuple[dict, str]:
+        """Initialize the state to debug from an exception.
```
Comment (skrawcz) [inline burr/tracking/client.py:83]:
```suggestion
        """Pull the serialized state from the logs. It's up to the user to make it a proper state values.
```

File: burr/core/application.py [hunk 2]
```diff
@@ -312,7 +320,13 @@ async def astep(self, inputs: Dict[str, Any] = None) -> Optional[Tuple[Action, d
             else:
                 result = await _arun_function(next_action, self._state, inputs=inputs)
                 new_state = _run_reducer(next_action, self._state, result, next_action.name)
-            new_state = new_state.update(**{PRIOR_STEP: next_action.name})
+            new_state = new_state.update(
```
[the remark this answers is not in the document]
Comment (skrawcz) [reply burr/core/application.py:323]: sure.

File: burr/core/application.py [hunk 3]
```diff
@@ -780,7 +794,7 @@ def with_tracker(
     ):
         """Adds a "tracker" to the application. The tracker specifies
         a project name (used for disambiguating groups of tracers), and plugs into the
-        Burr UI. Currently the only supported tracker is local, which takes in the params
+        Burr UI. Currently, the only supported tracker is local, which takes in the params
```
[the remark this answers is not in the document]
Comment (skrawcz) [reply burr/core/application.py:797]: Auto suggested by pycharm.

**Review by elijahbenizzy — APPROVED** [review]
LGTM

**Pull request merged**